### PR TITLE
fixed #10863  Update tableNameOrAlias to equalsIgnoreCase

### LINKED ISSUE #10863: Inconsistent case check problem
### Which version of ShardingSphere did you use?
4.1.1 & 5.0.0-RC1-SNAPSHOT

There is an inconsistency in case handling in class `TablesContext` and `ProjectionsContextEngine`, which needs to be corrected.

image

### PR DESCRIPTION
**Pull request by zoujianchao**
Fixes #10863

Update tableNameOrAlias to equalsIgnoreCase

### CODE REVIEW

**Comment by RaigorJiang** [comment]
Hi @zoujianchao
Thanks for you PR, but please notice:
1. Ref to #10863, please do not forget `TablesContext` has the same problem.
2. Please make your PR clean, do not hava too many files changed.
3. Please add unit test to check your modification.

Thank you!

**Comment by zoujianchao** [comment]
ok i will modify it

**Review by TeslaCN — CHANGES_REQUESTED** [review]
Hi @zoujianchao 
The 5.0.0-beta releasing is in progress and the branch 5.0.0-beta-release will not accept any commit. Please change the target branch to master.

**Comment by zoujianchao** [comment]
Ok, I have edited it to matser

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContext.java [hunk 1]
```diff
@@ -38,30 +38,30 @@
 @Getter
 @ToString
 public final class TablesContext {
-    
+
```
Comment (RaigorJiang) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContext.java:41]: Please do not modify the indent.   
Refer to https://shardingsphere.apache.org/community/cn/contribute/code-conduct/

Comment (zoujianchao) [reply shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContext.java:41]: Ok,This is my first pr, I'll notice it, Let me open pr up again

Comment (RaigorJiang) [reply shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContext.java:41]: Yes，everything is difficult at the beginning, please go on 👍

Comment (zoujianchao) [reply shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/segment/table/TablesContext.java:41]: thanks

**Pull request closed without merging**